R/bm-array-to-vector.R

```diff
@@ -10,7 +10,10 @@
 #'
 #' @importFrom purrr map flatten
 #' @export
-array_to_vector <- Benchmark("array_to_vector",
+array_to_vector <- Benchmark(
+  "array_to_vector",
+  version = "1.0.0",
```

Comment:
This might be too complicated to implement nicely, but one thing that this will do is keep the version for each case of the benchmark. And in practice I bet that most situations where we bump versions will actually be for a single case of a benchmark rather than all of them together. Any thoughts about it that's something we should account for?

I'm not really concerned about the additional versions (the numbers are free, after all!), but version bumps will + should disrupt history, and so we don't want to do that unless we actually need to, yeah?

Reply:
Hmmm, yeah that might work. I was also thinking separately, what if by default we ran version "highest" and then we defined in the actual benchmark code which case(s) have which numbers? 

So by default everything starts out as 1. Then if we change something for some specific case, we do something like we do with the validity bit and version 2 for that one case is selected when you run "highest" (we could decide separately if we want to keep around the code in if|else for running older versions too then)

Reply:
Ok, so my current thinking is that I want to close this PR and see if I can implement an opt-in case version via a function in the benchmark that takes in a case and appends a version to the params. The problem of how to version per benchmark is still there, but if we're not changing much, we can mostly ignore it.

I'm not sure yet whether the defaults written to the JSON (as a param, probably? negotiable) should be `1L`, `NULL`, or just not present; I might need to go look at conbench to see what would happen in each situation

Reply:
Sounds good. Another (possible) approach would be to wait for this until benchmarks is no longer providing all the case info. On the arrow side we don't yet have an urgent need for version numbers in arrow. But I'm happy to do it in benchmarks for now and flesh out more uncertainty while we are thinking about it.

Reply:
Yeah, I mostly want to make sure the code and logic paths are there to write JSON which can serve the purpose we need so we don't end up jumping through hoops to conform to what the endpoint needs